// components/404/index.js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Page404 extends Component {
  constructor(props) {
    super(props);
    this.state= {
      seconds: 5
    };
    this.interval = false;
  }

  componentDidMount(){
    let _this = this;
    this.interval = setInterval(()=>{
      _this.setState({
        seconds: _this.state.seconds - 1
      })
    }, 1000)
  }

  componentDidUpdate(){
    if (this.state.seconds === 0) {
      this.context.router.push('/')
    }
  }

  componentWillUnmount(){
    clearTimeout(this.interval);
  }

  render() {
    return (
      <div className="sign-up__block ov-h bg-none flex-1">
        <div className="container">
          <div className="row position-relative">
            <div className="squares__left"><span /></div>
            <div className="squares__right"><span /></div>
            <div className="col-10 col-md-6 col-lg-4 mx-auto">
              <div className="h2 mb-2 font-weight-light">404</div>
              <p className="mb-0 text-light-grey"> There is no route for your request <br/>
                You will be redirected in <span className='text-warning'> {this.state.seconds} </span> to main page
              </p>
            </div>
          </div>
        </div>
      </div>
    )
  }
}


Page404.contextTypes = {
  router: PropTypes.shape({}),
};

export default Page404;
