// components/layout/header/index.js
import React, { Component } from 'react';
import {Link} from "react-router";
import {userActions} from "_actions";
import {authentication} from "_reducers/auth.reducer";
import {connect} from "react-redux";
import { authActions } from '_actions'
import { Collapse } from 'reactstrap'
import Popup from "components/common/popup/popupTerms";
import header_logo from "img/logo-head.svg"

class Header extends Component {
  constructor(props){
    super(props);
    this.state = {
      mobMenuFlag: false
    }
  }
  componentDidMount(){
    let _this = this;
    window.addEventListener("resize", (event)=>{
      event.preventDefault();
      _this.sidebarFunc(window.innerWidth);
      return false;
    });

    this.sidebarFunc(window.innerWidth);
  }

  sidebarFunc(screenWidth){ // проверка расширения екрана
    if (screenWidth >= 768) {
      this.toggleMobMenu(true);
    }
  }

  toggleMobMenu(flag){ // тоглим состояние менюшки
    this.setState({
      mobMenuFlag: flag || !this.state.mobMenuFlag
    });
  }

  handleToggleMobMenu = (e)=>{ // тоглим состояние менюшки
    e.preventDefault();
    this.setState({
      mobMenuFlag: !this.state.mobMenuFlag
    });
    return false;
  };

  handleLogOutRequest = (e)=>{
    e.preventDefault();
    this.props.dispatch(authActions.logout());
  };
  render() {
    let {
      handleLogOutRequest,
      handleToggleMobMenu,
      state: {
        mobMenuFlag
      },
      props: {
        className
      }
    } = this;
    return (
      <header className={`header ${className}`}>

        <div className="container">
          <div className="row align-items-center">
            <div className="col-auto order-0">
              <Link className="text-dark header__logo" to='/'>
                  <img src={header_logo} alt="logo"/>
                {/*<span className="text-success">Cross</span>*/}
                {/*<span className="text-warning text-uppercase"> / capital </span>*/}
              </Link>
            </div>
            <div className="col-12 col-md-auto ml-md-auto order-4 order-md-1">
              <Collapse isOpen={mobMenuFlag}>
                <div className="d-flex align-items-center flex-wrap flex-column flex-md-row">
                  <menu className="header-menu d-flex align-items-center flex-wrap flex-column flex-md-row">
                    <li className="header-menu__item">
                      {/*<Link className="text-secondary font-weight-medium" to='/' >Investment plans</Link>*/}
                      <a className="text-secondary font-weight-medium" href='/#tradeLink' >Investment plans</a>
                    </li>
                    <li className="header-menu__item">
                      <Link className="text-secondary font-weight-medium" to='/faq'>FAQ</Link>
                    </li>
                    <li className="header-menu__item">
                      <Link className="text-secondary font-weight-medium" to='/support'>Support</Link>
                    </li>
                    <li className="header-menu__item">
                      <Link className="text-secondary font-weight-medium" to='/banners'>Banners</Link>
                    </li>
                    {

                      this.props.authentication.loggedIn ?
                        <li className="header-menu__item">
                          <Link className="text-secondary font-weight-medium" to='/profile'>Profile</Link>
                        </li> :
                        <li className="header-menu__item">
                          <Link className="text-secondary font-weight-medium" to='/auth/login'>Login</Link>
                        </li>
                    }
                    {
                      this.props.authentication.loggedIn ?
                        <li className="header-menu__item">
                          <a className="text-secondary font-weight-medium" href={''} onClick={handleLogOutRequest}>Logout</a>
                        </li> :
                        <li className="header-menu__item">
                            <a className="text-secondary font-weight-medium" href='/auth/sign-up/#signUpBlock' >Sign up</a>
                          {/*<Link className="text-secondary font-weight-medium" to='/auth/sign-up'>Sign up</Link>*/}
                        </li>
                    }

                  </menu>
                </div>
              </Collapse>
            </div>
            <div className="col-auto d-md-none order-1 ml-auto">
              <div className={`mob-menu__init ${mobMenuFlag ? 'toggled' : ''}`} onClick={handleToggleMobMenu}>
                <div className="action_btn">
                  <div className="first">x</div>
                  <div className="second" />
                  <div className="third" />
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

function mapStateToProps(state) {
  return {
    authentication: state.authentication,
  };
}

const _Header = connect(mapStateToProps)(Header);

export default _Header;
