// components/common/landing/index.js
import React, { Component } from 'react';
import { connect } from 'react-redux'
import TradePlanItem from "components/common/landing/tradePlan/index";
import ProfitCalculator from "components/profitCalculator";
import Slider from 'react-slick';
import Header from "components/layout/header/index";
import * as api from '_services'


import img_time1 from "img/statistic/time.png"
import img_time2 from "img/statistic/time2.png"
import img_statistic_stats from "img/statistic/stats.png"
import img_statistic_flows from "img/statistic/flows.png"
import img_statistic_persons from "img/statistic/persons.png"
import img_statistic_dollar_g from "img/statistic/dollar_g.png"
import img_statistic_dollar from "img/statistic/dollar.png"
import img_statistic_flows1 from "img/statistic/flows1.png"
import img_calc_img from "img/calc_img.jpg"
import img_features_certificate from "img/features/certificate.png"
import img_features_security from "img/features/security.png"
import img_features_company from "img/features/company.png"
import img_features_features from "img/features/features.png"
import img_tour from "img/tour.jpg"
import img_accept_payeer from "img/accept/payeer_c.png"
import img_accept_perf_money from "img/accept/perf_money_c.png"
import img_accept_adv_cash from "img/accept/adv_cash_c.png"

import img_pay_systems_adv_cash from "img/icons/pay-systems/adv_cash.png"
import img_pay_systems_pm from "img/icons/pay-systems/pm.png"
import img_pay_systems_py from "img/icons/pay-systems/py.png"

import img_slider_1 from "img/slider/bg1.png"
import img_slider_2 from "img/slider/slide2.jpg"
import img_slider_3 from "img/slider/slide3.jpg"
import img_cert from "img/cert.jpg"
import {Link} from "react-router";
import PopupSert from "components/common/popup/popupSert";




let PaymentSystemsIcons = {
  perfect_money: img_pay_systems_adv_cash,
  advanced_cash: img_pay_systems_pm,
  payeer: img_pay_systems_py
};

class Landing extends Component {
  constructor(props){
    super(props);
    this.state= {
      currentSlide: 1,
      slidesAmount: 1,
      lastOperations: false,
      needRenderSlider: true,
      statistic: false,
    };

    this.sliderSetting = {
      dots: false,
      arrows: false,
      autoplaySpeed: 4000,
        autoplay: true,
        infinite: true,
      speed: 900,
      slidesToShow: 1,
      slidesToScroll: 1,


      afterChange: index => {
        this.setState({
          currentSlide: index+1
        })
      },
      init: () => {
      }
    };

    this.refSlider = this.refSlider.bind(this);
  }

  componentWillMount(){
    let _this= this;

    api.tradePlansService.lastOperations()
      .then((res)=>{
        _this.setState({
          lastOperations: {
            operations: res.operations.concat(res.operations_face),
            user_plans: res.user_plans.concat(res.user_plans_face)
          },
        });
      });


    api.getProjectStatistic()
      .then((res)=>{
        _this.setState({
          statistic: res.data
        });
      })
  }

  refSlider(ref){
    if (ref) {
      this._slick = ref;
      this.setState({
        slidesAmount: ref.props.children.length
      })
    }
  }

  renderSlider(){
    let { _slick, sliderSetting } = this;
    let {
      currentSlide,
      slidesAmount
    } = this.state;
    return (
      <div id="main__slider position-relative">
        <Slider
          {...sliderSetting}
          ref={this.refSlider}
        >
          <div className="slide d-flex align-items-center red" style={{backgroundImage: `linear-gradient(0deg, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(${img_slider_1})`}}>
            <div className="container">
              <div className="row">
                <div className="col-12 col-md-8 col-lg-5">
                  <div className="h1 text-white slide__title font-weight-bold">Top class trading <br /> opportunity</div>
                  <p className="text-white mb-3">Build your future with Cross Capital</p>
                  <p className="text-white font-weight-bold mb-4">Get 135%, 205% profit from your investment.</p>
                  <a className="btn btn-lg btn-success" href='/auth/sign-up/#signUpBlock' >

                    Register
                  </a>
                  {/*<a className="btn btn-lg btn-success" href="#">Register</a>*/}
                </div>
              </div>
            </div>
          </div>
          <div className="slide d-flex align-items-center red" style={{backgroundImage: `linear-gradient(0deg, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(${img_slider_2})`}}>
            <div className="container">
              <div className="row">
                <div className="col-12 col-md-5">
                  <div className="h1 text-white slide__title">Top class trading <br /> opportunity</div>
                </div>
              </div>
            </div>
          </div>
            <div className="slide d-flex align-items-center red" style={{backgroundImage: `linear-gradient(0deg, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(${img_slider_3})`}}>
                <div className="container">
                    <div className="row">
                        <div className="col-12 col-md-5">
                            <div className="h1 text-white slide__title">Top class trading <br /> opportunity</div>
                        </div>
                    </div>
                </div>
            </div>
        </Slider>
        <div id="main__slider-nav">
          <button className="slick-prev slick-arrow" type="button" onClick={()=>_slick.slickPrev()}>Previous</button>
          <span className="position text-white">{currentSlide}/{slidesAmount}</span>
          <button className="slick-next slick-arrow" type="button" onClick={()=>_slick.slickNext()}>Next</button>
        </div>
      </div>
    )
  }

  componentWillUnmount(){
    this.setState({
      needRenderSlider: false
    })
  }
  render() {
    let { tradePlans } = this.props;
    let {
      needRenderSlider,
      lastOperations,
      statistic
    } = this.state;
    return (
      <div className="base-content">
        <div>
          <div className="mobile__menu" />
          <div className="main__slider-wrapper position-relative">
            <Header className={'header__main-page'}/>
            { needRenderSlider && this.renderSlider() }
          </div>
          <div className="about-project">
            <div className="container position-relative">
              {/*<div className="squares_top"/>*/}
              {/*<div className="squares_bottom"/>*/}
              <div className="row justify-content-between">
                <div className="col-12 mb-4">
                  <div className="h2 mb-3 text-dark font-weight-bold text-center">About us</div>
                </div>
              </div>
            </div>
          </div>
          <div className="about-project_bg-block">
            <div className="container text-white ">
              <div className="row">
                <div className="col-12 col-lg-7 order-1  order-lg-0 ui-fz-18">
                  <div className="lh-29  mb-2">
                    CrossCapital trades in the over-the-counter (otc) market known as "Forex", Forex is an act of simultaneous purchase of one currency and the sale of another, primarily for the purpose of speculation
                  </div>
                  <div className="lh-29  mb-4">
                    Also we have been investing in innovation projects since 2012. Before investing, we analyze a new project promptly and evaluate the risks. Having evaluated the maximum and minimum of the profit, we make an investing decision. Our team consists of professional financial experienced experts worked in investing companies all over the world
                  </div>
                  <div className="lh-29  mb-4">
                    <div className="text-center h4 font-weight-light"> Legality </div>
                    <div className="lh-27 ">
                      CrossCapital is a company that has built its business on an absolutely legal basis, in cooperation with the state and relevant state-controlled companies. With permits and licenses for all its activities,  CrossCapital is a company that has officially registered its commercial activities in accordance with international law.
                    </div>
                  </div>
                  <div className="lh-27">
                    <div className="text-center h4 font-weight-light"> Our Goal </div>
                    <p className='mb-3   '>
                      All of us, anyhow, strive to earn passive income, because only in this case a person has an opportunity to live a full life. Daily work does not always make it possible to earn a decent living, while passive income is not possible at all. That is why more and more people draw their attention to investing as the only way to get stable passive income.
                    </p>
                    <p className='mb-3 '>
                      If you are aiming for financial performance and long term-stability of your online incomes then we were made to work together. By joining LTD you will rely on successful years of experience of financial engineering. Our team contains highly experienced securities and trading professionals so you don't need to spend hours looking for the best opportunities because we do it for you.The purpose of our investment Fund is to provide regular profits for private investors with minimal risks.
                    </p>
                    <p className="">
                      Any investor can start to work and earn with us. Your Deposit works on a regular basis and you receive daily accruals with the ability to withdraw profits. <br/>
                      Join our team today and start to get high profit already tomorrow.
                    </p>
                  </div>
                </div>
                <div className="col-12 col-lg-5 order-0  order-lg-1 mb-4 mb-md-0">
                  <div className="right-side">
                    <div className="row">
                      <div className="col-12 col-sm-6 col-lg-12">
                        <div className="img-block d-inline-block mt-2 mb-3 mb-md-5">
                          <span className="squares" />
                          <div className="img-wrap"><img className="img-fluid" src={img_tour} alt="profit calc" /></div>
                        </div>
                      </div>
                      <div className="col-12 col-sm-6 col-lg-12">
                        <div className="text-white mt-4 ui-fz-14 lh-24">
                          LTD trades in the over-the-counter (otc) market known as "Forex", Forex is an act of simultaneous purchase of one currency and the sale of another, primarily for the purpose of speculation
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>

            </div>
          </div>
          <div className="features">
            <div className="title-line">
              <div className="container">
                <div className="row">
                  <div className="col-12 text-center mb-4">
                    <div className="h2 font-weight-bold">Company features</div>
                    <p className="mb-0 text-secondary mb-3">Things to make us proud</p>
                  </div>
                </div>
              </div>
            </div>
            <div className="features_bg">
              <div className="container">
                <div className="row feature-item__list">
                  <div className="col-12 col-md-6 feature-item-col">
                    <div className="feature-item h-100 d-flex flex-column">
                      <div>
                        <div className="feature__header d-inline-block"><span className="ico"><img src={img_features_features} alt="Features" /></span><span className="text font-weight-bold text-primary">Features</span></div>
                      </div>
                      <div className="feature__body flex-1 text-dark">
                        24/7 Trading<br/>
                        Your money works around the clock, all operations are available 24/7
                        Interest rates vary depending on the term, amount and interest payment frequency you choose.You can withdrawal your profit every business day
                        Your deposit is included in your payments.
                      </div>
                    </div>
                  </div>
                  <div className="col-12 col-md-6 feature-item-col">
                    <div className="feature-item h-100 d-flex flex-column">
                      <div>
                        <div className="feature__header d-inline-block"><span className="ico"><img src={img_features_company} alt="Company" /></span><span className="text font-weight-bold text-primary">Company</span></div>
                      </div>
                      <div className="feature__body flex-1 text-dark">
                        Cross Capital has been uniquely designed to give you a user-friendly & intelligible interface so that you can make the most of our investment platform without any hassle.Our Support service  is always ready to answer your questions
                      </div>
                    </div>
                  </div>
                  <div className="col-12 col-md-6 feature-item-col">
                    <div className="feature-item h-100 d-flex flex-column">
                      <div>
                        <div className="feature__header d-inline-block"><span className="ico"><img src={img_features_security} alt="Security" /></span><span className="text font-weight-bold text-primary">Security</span></div>
                      </div>
                      <div className="feature__body flex-1 text-dark">
                        Company website have EV SSL certificate offers the highest available levels of trust and authentication. The green address bar prominently displays our company name and provides highly visual assurance to customers that our site is secure.Personal data of users are stored and transmitted in an encrypted form
                      </div>
                    </div>
                  </div>
                  <div className="col-12 col-md-6 feature-item-col">
                    <div className="feature-item h-100 d-flex flex-column">
                      <div>
                        <div className="feature__header d-inline-block"><span className="ico"><img src={img_features_certificate} alt="Certificate" /></span><span className="text font-weight-bold text-primary">Certificate</span></div>
                      </div>
                      <div className="feature__body flex-1 ">
                        <div className="text-dark mb-2">
                          Our company is officially registered and has insurance in a well-established European insurance company
                        </div>
                          <a className="text-primary cursor-pointer font-weight-bold" id="popupCert"> + View Certificates </a>

                          <div id="modalPopupCert" className="modal-cert">
                              <div className="modal-content-cert">
                                  <span className="cursor-pointer closePopupCert">&times;</span>
                                <div className="row">
                                    <div className="col-md-12 d-flex justify-content-center">
                                        <img src={img_cert} className="cert-img" alt="certificate"/>
                                    </div>
                                </div>

                              </div>
                          </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="trade-plans ov-h" id="tradeLink">
            <div className="title-line">
              <div className="container">
                <div className="row">
                  <div className="col-12 text-center mb-3">
                    <div className="h2 font-weight-bold mb-4">Investment plans</div>
                  </div>
                </div>
              </div>
            </div>
            <div className="trade-plans_bg">
              <div className="container">
                <div className="row trade-plans__list">
                  {
                    tradePlans.map((elem,i)=>{
                      return <TradePlanItem key={i} plan={elem}/>
                    })
                  }
                </div>
              </div>
            </div>
          </div>
          <div className="affiliate-program">
            <div className="title-line">
              <div className="container">
                <div className="row">
                  <div className="col-12 text-center mb-3">
                    <div className="h2 font-weight-bold text-dark mb-4">Affilate Program 10%
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="affiliate-program_bg">
              <div className="container">
                <div className="row">
                  <div className="col-12 col-md-10 col-lg-auto mx-auto">
                    <div className="row">
                      <div className=" affiliate-programs__wrap mx-auto col-12">
                        <div className="affiliate-programs__list mb-3 d-flex flex-column flex-sm-row align-items-center justify-content-sm-between">
                          <div className="affiliate__item top mb-3 mb-sm-0">
                            <div className="circle"/>
                            <div className="value">7%</div>
                          </div>
                          <div className="affiliate__item medium mb-3 mb-sm-0">
                            <div className="circle"/>
                            <div className="value">2%</div>
                          </div>
                          <div className="affiliate__item start">
                            <div className="circle"/>
                            <div className="value">1%</div>
                          </div>
                          <div className="drawing"/>
                        </div>
                        <div className="lh-27 text-white text-center">
                          Invite someone to platform and get 7% of his deposit. If your invited users invite someone else you get 2% of their deposits
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="col-auto mx-auto ml-lg-auto mr-lg-0  text-white mt-3 mt-lg-0">
                    <div className="font-weight-bold ui-fz-24 mb-3 text-center text-lg-left">
                      Contact Us
                    </div>
                    <div className="lh-27 mb-1">
                      Admin 1 / Admin 2
                    </div>
                    <div className="lh-27 mb-3">
                      Whatsapp: <br/>
                      +228 93 84 53 80 /<br/>
                      +228 93 81 35 72
                    </div>
                    <a className="btn btn-success btn-lg" style={{width: '230px'}} href='/auth/sign-up/#signUpBlock' > Sign up </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="profit-calculator">
            <div className="title-line">
              <div className="container">
                <div className="row">
                  <div className="col-12 text-center">
                    <div className="h2 font-weight-bold text-dark mb-4">Profit calculator
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="profit-calculator_bg">
              <div className="container">
                <div className="row justify-content-between flex-wrap flex-lg-nowrap">
                  <div className="col-12 col-md-6 col-lg-auto">
                    <div className="left-side mr-md-4 mx-auto">
                      <ProfitCalculator />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>

          {
            statistic &&
            <div className="statistic">
              <div className="container title-line">
                <div className="row">
                  <div className="col-12 text-center">
                    <div className="mb-4">
                      <div className="h2 text-dark font-weight-bold">Company Statistics</div>
                      <p className="mb-0 text-secondary mb-3">Things to make us proud</p>
                    </div>
                  </div>
                </div>
              </div>

              <div className="statistic_bg">
                <div className="container">
                  <div className="row statistic__items_list">
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#007AFF'}}><img src={img_time1} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Started</span><span className="ui-fz-24" style={{color: '#007AFF'}}>{statistic.started}</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#5AC8FA'}}><img src={img_time2} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Running Days</span><span className="ui-fz-24" style={{color: '#5AC8FA'}}>{statistic.running_days} </span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#FF9500'}}><img src={img_statistic_stats} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Total Deposited</span><span className="ui-fz-24" style={{color: '#FF9500'}}>{statistic.total_deposited} €</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#FF9500'}}><img src={img_statistic_flows} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Total withdraw</span><span className="ui-fz-24" style={{color: '#FF9500'}}>{statistic.total_withdraw} €</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#FF3B00'}}><img src={img_statistic_persons} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Total accounts</span><span className="ui-fz-24" style={{color: '#FF3B00'}}>{statistic.total_accounts}</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#13D477'}}><img src={img_statistic_dollar_g} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Today deposited</span><span className="ui-fz-24" style={{color: '#13D477'}}>{statistic.today_deposited} €</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#30DBE5'}}><img src={img_statistic_dollar} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Last Deposit</span><span className="ui-fz-24" style={{color: '#30DBE5'}}>{statistic.last_deposited} €</span></div>
                      </div>
                    </div>
                    <div className="col-md-4 col-lg-3 col-sm-6 col-12">
                      <div className="statistic__item h-100">
                        <div className="ico-wrapper" style={{background: '#30DBE5'}}><img src={img_statistic_flows1} /></div>
                        <div className="text-body"><span className="ui-fz-14 d-block indicator">Last withdraw</span><span className="ui-fz-24" style={{color: '#30DBE5'}}>{statistic.last_withdraw} €</span></div>
                      </div>
                    </div>
                  </div>
                </div>

              </div>
            </div>
          }
          {
            lastOperations &&
            <div className="last-operations">
              <div className="title-line container">
                <div className="row">
                  <div className="col-12 text-center mb-4">
                    <div className="h2 font-weight-bold"> Last Operations </div>
                    <p className="mb-0 text-secondary mb-3">Last Operations of our users on platform</p>
                  </div>
                </div>
              </div>
              <div className="last-operations_bg">
                <div className="container">
                  <div className="row">
                    <div className="col-12 col-md-6 col-lg-5 col-lg-4  ml-lg-auto mb-4 m">
                      <div className="deposit-side text-white rounded">
                        <div className="title d-flex align-items-center h4">
                          <div className="d-block mr-4">
                            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19 21" width={19} height={21}>
                              <path fill="#00C73D" fillRule="evenodd" d="M12 0H7v11.4H0L9.5 21l9.5-9.6h-7V0z" />
                            </svg>
                          </div><span className="d-block text-success">Deposits</span>
                        </div>
                        <ul className="operation-list ui-fz-18 lh-22">
                          <marquee className="marquee" behavior="" direction="down" scrolldelay="50">
                            {
                              lastOperations.user_plans.map((plan , i) => {
                                return (
                                  <li className="operation__item d-flex align-items-center" key={i}>
                                    <div className="d-block"><img src={PaymentSystemsIcons[plan.payment_alias]} /></div>
                                    <div className="user-nick font-weight-bold flex-1">{plan.username}</div>
                                    <div className="text-right">{plan.initial_money}</div>
                                  </li>
                                );
                              })
                            }
                          </marquee>
                        </ul>
                      </div>
                    </div>
                    <div className="col-12 col-md-6 col-lg-5 col-lg-4 offset-lg-1 pr-md-0 mr-lg-auto">
                      <div className="withdraw-side text-white rounded">
                        <div className="title d-flex align-items-center h4">
                          <div className="d-block mr-4">
                            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 19 21" width={19} height={21}>
                              <path transform="rotate(180 9.5 10.5)" fill="#FF3B30" fillRule="evenodd" d="M19 11.4L9.5 21 0 11.4h7V0h5v11.4h7z" />
                            </svg>
                          </div><span className="d-block text-danger">Withdrawals</span>
                        </div>
                        <ul className="operation-list ui-fz-18 lh-22">
                          <marquee className="marquee" behavior="" direction="up" scrolldelay="50">
                            {
                              lastOperations.operations.map((operation , i) => {
                                return (
                                  <li className="operation__item d-flex align-items-center" key={i}>
                                    <div className="d-block"><img src={PaymentSystemsIcons[operation.payment_alias]} /></div>
                                    <div className="user-nick font-weight-bold flex-1">{operation.username}</div>
                                    <div className="text-right">{operation.sum}</div>
                                  </li>
                                );
                              })
                            }
                          </marquee>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>

              </div>
            </div>

          }
          <div className="accept__block">
            <div className="container">
              <div className="row">
                <div className="col-12 text-center">
                  <div className="ui-fz-24 font-weight-bold text-dark block-title">We Accept</div>
                </div>
              </div>
              <div className="row">
                <div className="col-12">
                  <div className="accept-list bg-white mx-auto d-flex justify-content-around flex-wrap">
                    <div className="accept-list__item"><img src={img_accept_payeer} alt="Payeer" /></div>
                    <div className="accept-list__item"><img src={img_accept_perf_money} alt="Perfect money" /></div>
                    <div className="d-none accept-list__item"><img src={img_accept_adv_cash} alt="Advcash" /></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    tradePlans: state.tradePlans,
    authentication: state.authentication
  };
}

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(Landing)
