// _constants/user.constants.js
export const userDataConstants = {
  REQUEST: 'USER_DATA_REQUEST',
  SUCCESS: 'USER_DATA_SUCCESS',
  FAILURE: 'USER_DATA_FAILURE',
};
export const userWalletsConstants = {
  REQUEST: 'USER_WALLETS_REQUEST',
  SUCCESS: 'USER_WALLETS_SUCCESS',
  FAILURE: 'USER_WALLETS_FAILURE',
};
export const userDepositsConstants = {
  REQUEST: 'USER_DEPOSITS_REQUEST',
  SUCCESS: 'USER_DEPOSITS_SUCCESS',
  FAILURE: 'USER_DEPOSITS_FAILURE',
  NEW_PLAN: 'USER_DEPOSITS_FAILURE',
};
export const userOperationsConstants = {
  REQUEST: 'USER_OPERATIONS_REQUEST',
  SUCCESS: 'USER_OPERATIONS_SUCCESS',
  FAILURE: 'USER_OPERATIONS_FAILURE',
  NEW_OPERATION: 'USER_NEW_OPERATION',
};
export const userReferralsConstants = {
  REQUEST: 'USER_REFERRALS_REQUEST',
  SUCCESS: 'USER_REFERRALS_SUCCESS',
  FAILURE: 'USER_REFERRALS_FAILURE',
};
export const userWithdrawCancelConstants = {
  REQUEST: 'USER_CANCEL_WITHDRAW_REQUEST',
  SUCCESS: 'USER_CANCEL_WITHDRAW_SUCCESS',
  FAILURE: 'USER_CANCEL_WITHDRAW_FAILURE',
};
export const withdrawRequestConstants = {
  REQUEST: 'USER_WITHDRAW_REQUEST',
  SUCCESS: 'USER_WITHDRAW_SUCCESS',
  FAILURE: 'USER_WITHDRAW_FAILURE',
};
export const activatePlanFromBalanceConstants = {
  REQUEST: 'ACTIVATE_PLAN_FROM_BALANCE_REQUEST',
  SUCCESS: 'ACTIVATE_PLAN_FROM_BALANCE_SUCCESS',
  FAILURE: 'ACTIVATE_PLAN_FROM_BALANCE_FAILURE',
};
export const activatePlanConstants = {
  REQUEST: 'ACTIVATE_PLAN_REQUEST',
  SUCCESS: 'ACTIVATE_PLAN_SUCCESS',
  FAILURE: 'ACTIVATE_PLAN_FAILURE',
};

export const changeUserDataConstants = {
  REQUEST: 'CHANGE_USER_DATA_REQUEST',
  SUCCESS: 'CHANGE_USER_DATA_SUCCESS',
  FAILURE: 'CHANGE_USER_DATA_FAILURE',
};


export const changeUserPasswordConstants = {
  REQUEST: 'CHANGE_USER_PASSWORD_REQUEST',
  SUCCESS: 'CHANGE_USER_PASSWORD_SUCCESS',
  FAILURE: 'CHANGE_USER_PASSWORD_FAILURE',
};
