// components/layout/profile/index.js
import React, {Component} from 'react';
import helperFunctions from "_helpers/functions";
import ProfileSidebar from 'components/layout/profile/sidebar'
import ProfileHeader from "components/layout/profile/header/index";
import {authActions} from "_actions/auth.actions";
import {connect} from "react-redux";
import PropTypes from 'prop-types'
import {userActions} from "_actions";

import { Modal } from 'reactstrap';


class ProfileWrapper extends Component {
  constructor(props){
    super(props);
    this.state = {
      modalCalc: false
    };

    this.toggleModal = this.toggleModal.bind(this);
  }
  componentWillMount() {
    if (!this.props.authentication.loggedIn) { // если не залогинен
      this.context.router.push('/auth/login') // редирект на логин
    } else {
      // тянем данные
      this.props.fetchUserData(); // о пользователе,
      this.props.fetchUserWallets(); // о  балансе его кошельков
    }
  }
  componentWillReceiveProps(nextProps, nextState) {
    if (!nextProps.authentication.loggedIn) { // если не залогинен
      this.context.router.push('/auth/login') // редирект на логин
    }
  }
  toggleModal(modal){
    this.setState({
      [modal]: !this.state[modal]
    })
  }
  render() {
    let {
      toggleModal
    } = this;
    let {
      modalCalc
    } = this.state;
    return (
      <div className="profile__wrapper d-flex h-100 ov-h position-relative">
        <svg display="none" xmlns="http://www.w3.org/2000/svg">
          <symbol id="svg_arr_tdl" viewBox="0 0 12 10" width={12} height={10}>
            <path transform="rotate(180 6 5)" fillRule="evenodd" d="M0 4.667l4.667 4.666V6.6C7.999 6.6 10.333 7.667 12 10c-.667-3.333-2.667-6.667-7.333-7.333V0L0 4.667z" />
          </symbol>
          <symbol id="svg_arr_d" viewBox="0 0 12 8" width={12} height={8}>
            <path transform="translate(1 1)" d="M5 5l-.707.707.707.707.707-.707L5 5zM-.707.707l5 5 1.414-1.414-5-5L-.707.707zm6.414 5l5-5L9.293-.707l-5 5 1.414 1.414z" />
          </symbol>
          <symbol id="svg_persons" viewBox="0 0 21 14" width={21} height={14}>
            <path fillRule="evenodd" d="M14.297 7.986c.824 0 1.739.129 2.745.385 1.114.305 2.021.698 2.723 1.179.823.577 1.235 1.227 1.235 1.948V14H15.28v-2.502c0-1.363-.625-2.517-1.876-3.464.214-.032.511-.048.892-.048zm-7.595 0c.824 0 1.731.129 2.723.385 1.113.305 2.013.698 2.699 1.179.823.577 1.235 1.227 1.235 1.948V14H0v-2.502c0-.721.412-1.37 1.235-1.948.702-.481 1.61-.874 2.722-1.179 1.007-.256 1.922-.385 2.745-.385zm0-1.972c-.503 0-.976-.137-1.418-.41a2.98 2.98 0 0 1-1.052-1.106 3.015 3.015 0 0 1-.39-1.491c0-.53.13-1.026.39-1.492A2.98 2.98 0 0 1 5.284.41 2.656 2.656 0 0 1 6.702 0 2.6 2.6 0 0 1 8.11.409c.435.273.778.637 1.03 1.094.251.457.377.959.377 1.504S9.39 4.053 9.138 4.51A2.947 2.947 0 0 1 8.11 5.605a2.6 2.6 0 0 1-1.407.409zm7.596 0c-.503 0-.976-.137-1.418-.41a2.98 2.98 0 0 1-1.053-1.106 3.015 3.015 0 0 1-.389-1.491c0-.53.13-1.026.39-1.492A2.98 2.98 0 0 1 12.88.41a2.657 2.657 0 0 1 2.836 0 2.98 2.98 0 0 1 1.052 1.106c.26.466.39.963.39 1.492 0 .53-.13 1.026-.39 1.491a2.98 2.98 0 0 1-1.052 1.107 2.657 2.657 0 0 1-1.418.409z" />
          </symbol>
          <symbol id="svg_dollar" viewBox="0 0 12 21" width={12} height={21}>
            <path d="M6.46 9.217c-2.676-.689-3.537-1.4-3.537-2.509 0-1.271 1.191-2.158 3.183-2.158 2.098 0 2.876.992 2.947 2.45h2.605c-.082-2.007-1.32-3.85-3.784-4.445V0H4.338v2.52C2.05 3.01.212 4.48.212 6.732c0 2.695 2.252 4.036 5.54 4.818 2.947.7 3.537 1.727 3.537 2.812 0 .805-.578 2.088-3.183 2.088-2.428 0-3.383-1.073-3.513-2.45H0c.141 2.555 2.075 3.99 4.338 4.468V21h3.536v-2.508C10.173 18.06 12 16.742 12 14.35c0-3.313-2.864-4.445-5.54-5.133z" />
          </symbol>
          <symbol id="svg_stats" viewBox="0 0 17 17" width={17} height={17}>
            <path d="M6.375 17h4.25V0h-4.25v17zM0 17h4.25V8.5H0V17zM12.75 5.312V17H17V5.312h-4.25z" />
          </symbol>
          <symbol id="svg_settings" viewBox="0 0 20 20" width={20} height={20}>
            <path fillRule="evenodd" d="M10 13.51c.642 0 1.242-.16 1.802-.481.56-.32 1.004-.75 1.333-1.286A3.275 3.275 0 0 0 13.628 10c0-.625-.164-1.21-.493-1.755a3.65 3.65 0 0 0-1.333-1.298c-.56-.32-1.16-.48-1.802-.48-.641 0-1.238.16-1.789.48-.551.32-.991.753-1.32 1.298A3.336 3.336 0 0 0 6.397 10c0 .625.165 1.206.494 1.743.329.537.769.965 1.32 1.286.551.32 1.148.48 1.79.48zm7.675-2.524l2.147 1.634c.099.08.157.185.173.313a.555.555 0 0 1-.074.36l-2.073 3.462a.448.448 0 0 1-.26.216.56.56 0 0 1-.357-.024l-2.542-.986c-.658.465-1.242.794-1.752.986l-.37 2.62a.607.607 0 0 1-.185.313.453.453 0 0 1-.309.12H7.927a.485.485 0 0 1-.469-.433l-.395-2.62c-.69-.272-1.267-.6-1.727-.986l-2.567.986c-.263.112-.469.048-.617-.192L.079 13.293a.555.555 0 0 1-.074-.36.464.464 0 0 1 .173-.313l2.171-1.634A7.323 7.323 0 0 1 2.3 10c0-.433.017-.761.05-.986L.177 7.38a.464.464 0 0 1-.173-.313.555.555 0 0 1 .074-.36l2.073-3.462c.148-.24.354-.304.617-.192l2.567.985a8.02 8.02 0 0 1 1.727-.985l.395-2.62A.485.485 0 0 1 7.927 0h4.146c.115 0 .218.04.309.12.09.08.152.184.185.313l.37 2.62a6.45 6.45 0 0 1 1.752.985l2.542-.985a.56.56 0 0 1 .358-.024.448.448 0 0 1 .259.216l2.073 3.462c.066.112.09.232.074.36a.464.464 0 0 1-.173.313l-2.147 1.634c.033.225.05.553.05.986 0 .433-.017.761-.05.986z" />
          </symbol>
          <symbol id="svg_faq" viewBox="0 0 17 19">
            <path d="M15.111 1.9h-3.948C10.767.798 9.728 0 8.5 0 7.272 0 6.233.798 5.837 1.9H1.889A1.9 1.9 0 0 0 0 3.8v13.3A1.9 1.9 0 0 0 1.889 19H15.11A1.9 1.9 0 0 0 17 17.1V3.8a1.9 1.9 0 0 0-1.889-1.9zM8.5 1.9a.95.95 0 0 1 0 1.9.95.95 0 0 1 0-1.9zm1.889 13.3H3.778v-1.9h6.61v1.9zm2.833-3.8H3.778V9.5h9.444v1.9zm0-3.8H3.778V5.7h9.444v1.9z" />
          </symbol>
        </svg>
        <div className="container">
          <div className="row h-100">
            <ProfileSidebar/>
            <div className="col-12 col-lg-9">
              <div className="profile-body">
                <div className="container">
                  <div className="row">
                    <ProfileHeader/>
                  </div>
                  {this.props.children}
                </div>
              </div>
            </div>

            <div className="modal modal_profile fade" id="m_calculate" tabIndex={-1} role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
              <div className="modal-dialog modal-dialog-centered" role="document">
                <div className="modal-content">
                  <div className="modal-header">
                    <h5 className="modal-title">Calculate <br /> something</h5>
                    <button className="close" type="button" data-dismiss="modal" aria-label="Close">×</button>
                  </div>
                  <div className="modal-body">
                    <div className="profit-calculator p-0">
                      <p className="mb-0 text-light-grey ui-fz-18">Deposit amount</p>
                      <label className="value-amount d-block">
                        <div className="input" data-currency="$">
                          <input className="form-control w-100" id="calc__input" type="text" defaultValue={100} />
                        </div>
                        <div id="calc__slider" />
                      </label>
                      <dl className="ui-fz-18 text-dark mb-0">
                        <div className="line net-profit d-flex">
                          <dt className="text-light-grey">Invest plan:</dt>
                          <dd className="flex-1 text-orange font-weight-bold" id="calc__plan">Medium</dd>
                        </div>
                        <div className="calc__divider" style={{margin: '15px 0 23px'}} />
                        <div className="line net-profit d-flex">
                          <dt className="text-light-grey">Net profit:</dt>
                          <dd className="flex-1" id="calc__net-profit">150%</dd>
                        </div>
                        <div className="line period d-flex">
                          <dt className="text-light-grey">Period:</dt>
                          <dd className="flex-1" id="calc__period">100 Days</dd>
                        </div>
                        <div className="line amount d-flex">
                          <dt className="text-light-grey">Amount</dt>
                          <dd className="flex-1" id="calc__amount">
                            From
                            <span>
                          300
                        </span>to
                            <span>
                          2000
                        </span>Euro
                          </dd>
                        </div>
                        <div className="calc__divider" style={{margin: '19px 0 22px'}} />
                        <div className="line profit__summary d-flex">
                          <dt className="ui-fz-22 text-dark">Your profit:</dt>
                          <dd className="flex-1 ui-fz-24 text-success font-weight-bold" id="calc__profit__summary">$1500</dd>
                        </div>
                        <div className="line profit__per-day d-flex mb-0">
                          <dt className="text-light-grey">Profit per day:</dt>
                          <dd className="flex-1 text-light-grey" id="calc__profit__per-day">$15</dd>
                        </div>
                      </dl>
                    </div>
                  </div>
                  <div className="modal-footer">
                    <button className="btn btn-success btn-lg w-100" type="button" data-dismiss="modal">Calculate</button>
                  </div>
                </div>
              </div>
            </div>


          </div>
        </div>
      </div>
    );
  }
}

ProfileWrapper.contextTypes = {
  router: PropTypes.shape({}),
};

function mapStateToProps(state) {
  return {
    authentication: state.authentication
  };
}

const connectedProfileWrapper = connect(mapStateToProps,{
  logout: authActions.logout,
  fetchUserData: userActions.fetchUserData,
  fetchUserWallets: userActions.fetchUserWallets
})(ProfileWrapper);
export default connectedProfileWrapper;
