// components/profile/profileSettings/index.js
import React,{ Component } from 'react'
import { Link } from "react-router";
import { userActions } from "_actions";
import { connect } from "react-redux";
import PropTypes from 'prop-types'
import { toFloatNumbers } from '_helpers'
import { tradePlans } from "_reducers/trade.plans.reducer";
import {Nav, NavItem, NavLink, TabContent, TabPane} from "reactstrap";

import ProfileSettingsPassword from './Password'
import ProfileSettingsPersonalInfo from './PersonaInfo'
import ProfileSettingsWallets from './Wallets'

class ProfileSettings extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      activeTab: '1'
    };
  }

  toggle(tab) {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }
  render() {
    return (
      <div className='row'>
        <div className="col-12">
          <div className="h2 mb-4 font-weight-light">Settings</div>
          <div className="profile-settings-wrap">
            <Nav tabs className={'border-0 profile-settings-wrap__tab-menu'}>
              <NavItem>
                <NavLink
                  className={`cu-pointer ui-fz-18 ${this.state.activeTab === '1' ? 'active' :''} border-0`}
                  onClick={() => { this.toggle('1'); }}
                >
                  Personal info
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  className={`cu-pointer ui-fz-18 ${this.state.activeTab === '2' ? 'active' :''} border-0`}
                  onClick={() => { this.toggle('2'); }}
                >
                  Password
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  className={`cu-pointer ui-fz-18 ${this.state.activeTab === '3' ? 'active' :''} border-0`}
                  onClick={() => { this.toggle('3'); }}
                >
                  Wallet
                </NavLink>
              </NavItem>
            </Nav>
            <TabContent activeTab={this.state.activeTab} className={'bg-white profile-settings__tabs'}>
              <TabPane tabId="1">
                <ProfileSettingsPersonalInfo/>
              </TabPane>
              <TabPane tabId="2">
                <ProfileSettingsPassword/>
              </TabPane>
              <TabPane tabId="3">
                <ProfileSettingsWallets/>
              </TabPane>
            </TabContent>
          </div>
        </div>
      </div>
    )
  }
}


ProfileSettings.contextTypes = {
  router: PropTypes.shape({}),
};

function mapStateToProps(state) {
  const { loggingIn } = state.authentication;
  return {
    userOperations: state.userOperations
  };
}

const _ProfileSettings = connect(mapStateToProps,{
  fetchUserWithdrawals: userActions.fetchUserWithdrawals,
  cancelUserWithdraw: userActions.cancelUserWithdraw,
  fetchUserWallets: userActions.fetchUserWallets
})(ProfileSettings);

export default _ProfileSettings;
