// components/common/faq/index.js
import React, { Component } from 'react';
import {Collapse} from "reactstrap";
import Header from "components/layout/header/index";

class Faq extends Component {
  constructor(props) {
    super(props);
    this.state = {
      faqFlag: -1,
    };

    this.faqList = [
      // ABOUT THE COMPANY
      {
        title: 'What is Crosslink Capital?',
        message: 'This is the investment platform of  Crosslink Capital.  Crosslink Capital is an innovative project in the world of trading and IT development, providing a set of analytical tools and forecasting for trading on crypto-exchange exchanges, completely based on artificial intelligence.'
      },
      {
        title: 'How many years has the company existed on the market?',
        message: 'Crosslink Capital has been operating since 2012. The platform was launched in 2018. '
      },
      {
        title: 'Does the company conduct official activities?',
        message: 'Our company is officially registered and has insurance in a well-established European insurance company'
      },
      {
        title: 'What will happen to my deposit if the company for some reason closes?',
        message: 'The company has developed a development strategy until 2020, which will result in the creation of a trading strategies generator, a screen reader for cryptograms on neural networks  Until then, all participants will receive profit according to the selected investment plan. . In case of early closure of the project, for whatever reasons, the company undertakes to return the deposits deposited by investors, minus the already paid funds.'
      },
      // REGISTRATION
      {
        title: 'I want to become an investor, how can I do this?',
        message: 'First, you need to complete the registration process and familiarize yourself with our investment plans. Selecting the most attractive option, make a deposit and start get profit.'
      },
      {
        title: 'Can I create more than one account?',
        message: 'Each investor can register only one account. Creation of several accounts does not contradict the company rules only if the accounts are not united by referral links. Otherwise, all accounts are blocked.'
      },
      {
        title: 'Is it possible to become an investor if I am less than 18 years old?',
        message: 'No, cooperation with the company is possible only after reaching the age of majority.'
      },
      {
        title: 'I forgot my password?',
        message: 'Use the password recovery feature. Follow all password recovery points. The new password will be sent to the e-mail you specified during registration.'
      },
      {
        title: 'I forgot my login?',
        message: 'Contact support and specify the e-mail address you used when registering. Login will be sent to your e-mail.'
      },
      {
        title: 'Can I change my password?',
        message: 'Yes, you can change your password in your personal profile. '
      },
      {
        title: 'What payment systems do you work with?',
        message: 'We work with payment systems: Perfect Money, AdvCash, Payeer,'
      },
      {
        title: 'I want to change the purse, to which I will withdraw money, how to do it?',
        message: 'You can change the purse number on your own in the section - Payment details'
      },
      {
        title: 'What Commission do you charge for withdrawal of funds?',
        message: 'The company does not charge any fees, except those provided for in the electronic payment systems.2%from amount of the withdrawal.\n'
      },
      {
        title: 'Can I open more than one deposit?',
        message: 'You can open as many deposit as you like.'
      },
      {
        title: 'Can I use the funds accrued in my account to open a new deposit or replenish the current one?',
        message: 'The received profit is on the balance of the personal cabinet, you can  only create a new deposit '
      },
      {
        title: 'What is the minimum investment amount?',
        message: 'The minimum replenishment is 10 euro'
      },
      {
        title: 'How often will the profit be calculated?',
        message: 'From the moment the deposit is opened, profits are accrued daily after 24 hours.'
      },
      {
        title: 'Do you return the deposit amount?',
        message: 'The deposit amount is included in the payments.'
      },
      {
        title: 'What is the minimum and maximum amount I can request for withdrawal?',
        message: 'The minimum amount available for withdrawal  is 10 euro and no  limits for maximum. You can withdrawal your funds from Monday to Friday.'
      },
      {
        title: 'How many time is processing of applications for withdrawal of funds?',
        message: 'Typically, the money is transferred to your wallet within from 15 minutes to 24 hours'
      },
      // AFFILIATE PROGRAM
      {
        title: 'Do you have an affiliate program in your company?',
        message: 'Yes, our company has an affiliate program,more precisely, 7% level 1, 2% level 2. and 1% level 3.'
      },
      {
        title: 'Can I be an affiliate without making a deposit?',
        message: 'No, to have an active Deposit is not required. You have to just spread your referral link in any available to you.\n'
      },
      {
        title: 'What you should know about affiliate Commission?',
        message: 'Your affiliate reward will come each time your partner make deposit.You can make a deposit and strat earning,you can withdrawal referal funds if your account is active.'
      },
      {
        title: 'How to track the results of investment?',
        message: 'View information about your affiliate income you can in your  personal account'
      },
      // ADDITIONAL INFORMATION
      {
        title: 'I still have questions what should I do?',
        message: 'If You have not found the answer to the question in the FAQs, email support.'
      },
    ];

    this.toggleFaq = this.toggleFaq.bind(this);
  }

  toggleFaq(i){
    let { faqFlag } = this.state;
    let flag = -1;
    if (faqFlag !== i ) {
      flag = i;
    }
    this.setState({
      faqFlag: flag
    });
  }
  render(){
    let {
      faqList,
      toggleFaq
    } = this;
    let {
      faqFlag
    } = this.state;
    return (
      <div className="base-content">
        <Header/>
        <div className='flex-1'>
          <div className="mobile__menu" />
          <div className="faq-wrapper">
            <div className="container">
              <div className="row">
                <div className="col-12 col-md-5 col-lg-4">
                  <div className="faq__left-block"><span className="squares" />
                    <h2 className="font-weight-light mb-4">Frequently Asked Questions</h2>
                    <p className="text-light-grey lh-27">We had the ambition to make a platform that would help people make a sustainable income source through modern techniques</p>
                  </div>
                </div>
                <div className="col-12 col-md-7 col-lg-8">
                  <ul className="faq__questions-list">
                    {
                      faqList.map((faq, i) => {
                        return(
                          <li
                            className="faq__question"
                            key={`faq_${i}`}
                          >
                            <div
                              className="toggle__header faq__question-title ui-fz-22 lh-29 text-dark d-flex align-items-center justify-content-between"
                              onClick={()=>{ toggleFaq(i) }}
                            >
                              <div className="text">{faq.title}</div>
                              <div className="toggle-init__ico">+</div>
                            </div>
                            <Collapse isOpen={faqFlag > -1 && i === faqFlag}>
                              <div
                                className="toggle__body faq__question-body text-light-grey"
                              >
                                {faq.message}
                              </div>
                            </Collapse>
                          </li>
                        )
                      })
                    }
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Faq;
